docs/docusaurus/multitenancy/tenant-onboarding.md

```diff
@@ -0,0 +1,165 @@
+| `POST /iam/entities`              | Create a new entity on PRISM Agent         | Administrator |
+| `POST /iam/apikey-authentication` | Create a new authentication for the entity | Administrator |
+| `GET /did-registrar/dids`         | List the DIDs inside the wallet            | Tenant        |
+
+## Administrator interactions
+
+### 1. Check the existing wallets
+
+When running PRISM Agent using the configurations above, the Agent should start with an empty state.
+Listing wallets on it should return empty results.
+
+```bash
+curl -X 'GET' \
+  'http://localhost:8080/prism-agent/wallets' \
+  -H 'accept: application/json' \
+  -H 'x-admin-api-key: my-admin-token'
+```
+
+Response Example:
+
+```json
+{
+  "self": "/wallets",
+  "kind": "WalletPage",
+  "pageOf": "/wallets",
+  "contents": []
+}
+```
+
+### 2. Create a new wallet
+
+The wallet can be created using a `POST /wallets` endpoint.
+This wallet is going to act as a container for the tenant's assets (DIDs, VCs, Connections, etc.).
+The wallet seed may be provided during the wallet creation or omitted to let the Agent generate one randomly.
+
+
+```bash
+curl -X 'POST' \
+  'http://localhost:8080/prism-agent/wallets' \
+  -H 'accept: application/json' \
+  -H 'x-admin-api-key: my-admin-token \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "seed": "c9994785ce6d548134020f610b76102ca1075d3bb672a75ec8c9a27a7b8607e3b9b384e43b77bb08f8d5159651ae38b98573f7ecc79f2d7e1f1cc371ce60cf8a",
+    "name": "my-wallet"
+  }'
+```
+
+Response Example:
+
+```json
+{
+  "id": "99734c87-5c9d-4697-b5fd-dea4e9590ba7",
+  "name": "my-wallet",
+  "createdAt": "2023-01-01T00:00:00Z",
+  "updatedAt": "2023-01-01T00:00:00Z"
+}
+```
+
+### 3. Create a new entity
+
+A new entity can be created to represent a tenant.
+To create a new entity, send a `POST` request to the `/iam/entities` endpoint with the following parameters:
+
+```bash
+curl -X 'POST' \
+  'http://localhost:8080/prism-agent/iam/entities' \
+  -H 'accept: application/json' \
+  -H 'x-admin-api-key: my-admin-token' \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "name": "John",
+    "walletId": "99734c87-5c9d-4697-b5fd-dea4e9590ba7"
+  }'
+```
+
+Response Example:
+
+```json
+{
+  "kind": "Entity",
+  "self": "/iam/entities/10000000-0000-0000-0000-000000000000",
+  "id": "10000000-0000-0000-0000-000000000000",
+  "name": "John",
+  "walletId": "99734c87-5c9d-4697-b5fd-dea4e9590ba7",
+  "createdAt": "2023-09-01T14:00:38.760045Z",
+  "updatedAt": "2023-09-01T14:00:38.760047Z"
+}
+```
+
+### 4. Register `apikey` authentication method
+
+With the new tenant now equipped with both a wallet and an entity,
+the final step involves setting up the entity's authentication method.
+Once this step is completed, the administrator should provide the tenant with an `apikey`, granting them access to utilize the Agent.
+
+```bash
+curl -X 'POST' \
+  'http://localhost:8080/prism-agent/iam/apikey-authentication' \
```

Comment:
We need to align all URLs we use in the examples.
Sometimes we use `localhost:8080/wallets`, and sometimes `localhost/prism-agent/wallets.`

Reply:
It should always be `http://localhost:8080/prism-agent` with the port, for now, I believe.

Reply:
It depends on how are you starting the agent:
If it's a plain docker - 8080
If it's a docker-compose with APISIX proxy, Vault, and DB - it's either localhost:80 or localhost:8080 (in the case when the 8080 is exposed.